core/src/main/java/com/sap/ai/sdk/core/DestinationResolver.java

```diff
@@ -0,0 +1,106 @@
+package com.sap.ai.sdk.core;
+
+import static com.sap.cloud.sdk.cloudplatform.connectivity.OnBehalfOf.TECHNICAL_USER_PROVIDER;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.sap.cloud.environment.servicebinding.api.DefaultServiceBindingAccessor;
+import com.sap.cloud.environment.servicebinding.api.DefaultServiceBindingBuilder;
+import com.sap.cloud.environment.servicebinding.api.ServiceBindingMerger;
+import com.sap.cloud.environment.servicebinding.api.ServiceIdentifier;
+import com.sap.cloud.sdk.cloudplatform.connectivity.DefaultHttpDestination;
+import com.sap.cloud.sdk.cloudplatform.connectivity.HttpDestination;
+import com.sap.cloud.sdk.cloudplatform.connectivity.ServiceBindingDestinationLoader;
+import com.sap.cloud.sdk.cloudplatform.connectivity.ServiceBindingDestinationOptions;
+import java.util.HashMap;
+import java.util.List;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import lombok.extern.slf4j.Slf4j;
+
+/** Utility class to resolve the destination pointing to the AI Core service. */
+@Slf4j
+class DestinationResolver {
+
```

Comment:
How about: 
```suggestion
  private ServiceBindingAccessor accessor = DefaultServiceBindingAccessor.getInstance();
```
for easier unit testing?

Reply:
Yeah sure, we didn't test this in the past and I wasn't planning to add test.

I'm not happy with the class in general, but I didn't want to change it as part of this ticket / PR.